recipes/notcurses/all/conanfile.py

```diff
@@ -0,0 +1,41 @@
+from conans import ConanFile, CMake, tools
+
+
+class NotcursesConan(ConanFile):
+    name = "notcurses"
+    description = "a blingful TUI/character graphics library"
+    topics = ("graphics", "curses", "tui", "console", "ncurses")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://nick-black.com/dankwiki/index.php/Notcurses"
+    license = "Apache-2.0"
+    exports_sources = ["CMakeLists.txt"]
+    settings = "os", "compiler", "build_type", "arch"
+    requires = "openimageio/2.2.7.0", "libunistring/0.9.10"
+    options = {"shared": [True, False]}
+    default_options = {"shared": True,
+                       "with_doctest": True}
+    generators = "cmake"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure(source_folder="notcurses")
+        cmake.build()
+
+        # Explicit way:
+        # self.run('cmake %s/notcurses %s'
+        #          % (self.source_folder, cmake.command_line))
+        # self.run("cmake --build . %s" % cmake.build_config)
```

Comment:
```suggestion
    @property
    def _source_subfolder(self):
        return "source_subfolder"

    def requirements(self):
        self.build_requires("libunistring/0.9.10")
        self.build_requires("openimageio/2.2.7.0")

    def build_requirements(self):
        self.build_requires("pkgconf/1.7.3")

    def source(self):
        tools.get(**self.conan_data["sources"][self.version])
        extracted_dir = self.name + "-" + self.version
        os.rename(extracted_dir, self._source_subfolder)

    def build(self):
        cmake = CMake(self)
        cmake.configure()
        cmake.build()

```

Reply:
i missed this somehow, but i've since synthesized most of it on my own =]. thank you!